spec/black_box/black_box_spec.rb

```diff
@@ -0,0 +1,77 @@
+require "rails_helper"
+
+RSpec.describe BlackBox do
+  let!(:function_caller) { double }
+
+  describe "#article_hotness_score" do
+    let!(:article) { create(:article, published_at: Time.current) }
+
+    it "calls function caller" do
+      allow(function_caller).to receive(:call).and_return(5)
+      described_class.article_hotness_score(article, function_caller)
+      expect(function_caller).to have_received(:call).once
+    end
+
+    it "doesn't fail when function caller returns nil" do
+      allow(function_caller).to receive(:call).and_return(nil)
+      described_class.article_hotness_score(article, function_caller)
+    end
+
+    it "returns the correct value" do
+      article.update_column(:score, 99)
+      allow(function_caller).to receive(:call).and_return(5)
+      # recent bonuses (28 + 31 + 80 + 395 + 330 + 330 = 1194)
+      # + score (99)
+      # + value from the function caller (5)
+      score = described_class.article_hotness_score(article, function_caller)
+      expect(score).to eq(1298)
+    end
+  end
+
+  describe "#comment_quality_score" do
+    let(:comment) { create(:comment, commentable: create(:article), body_markdown: "```#{'hello, world! ' * 20}```") }
+
+    before do
+      reaction = create(:reaction, reactable: comment)
+      reaction.update_column(:points, 20)
```

Comment:
you can just say create(:reaction, reactable: comment, points: 20) I guess

Reply:
points will be assigned in the `before_save` callback then and the specified value will be overwritten